pkg/webhook/storageclass_test.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	v1 "k8s.io/api/admission/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func TestMutateStorageClass(t *testing.T) {
+	storageClassName := "filestore-multishare"
+	labelName := "multishare-label"
+
+	testCases := []struct {
+		name         string
+		storageClass *storagev1.StorageClass
+		operation    v1.Operation
+		shouldAdmit  bool
+		patch        string
+		msg          string
+	}{
+		{
+			name:        "Update should not be allowed",
+			operation:   v1.Update,
+			shouldAdmit: false,
+			msg:         fmt.Errorf("resource %v is immutable", &StorageClassV1GVR).Error(),
+		},
+		{
+			name: "create with non-multishare should be allowed",
+			storageClass: &storagev1.StorageClass{
+				ObjectMeta:  metav1.ObjectMeta{Name: storageClassName},
+				Provisioner: FilestoreCSIDriver,
+			},
+			operation:   v1.Create,
+			shouldAdmit: true,
+		},
+		{
+			name: "create with other provisioner should be allowed",
+			storageClass: &storagev1.StorageClass{
+				ObjectMeta:  metav1.ObjectMeta{Name: storageClassName},
+				Provisioner: "pd.csi.storage.gke.io",
+			},
+			operation:   v1.Create,
+			shouldAdmit: true,
+		},
+		{
+			name: "create with multishare but default tier should not be allowed",
+			storageClass: &storagev1.StorageClass{
+				ObjectMeta:  metav1.ObjectMeta{Name: storageClassName},
+				Provisioner: FilestoreCSIDriver,
+				Parameters: map[string]string{
+					"multishare": "true",
```

Comment:
test cases for "multishare" : "" and "multishare" : "blah". it should fail

Reply:
done